src/app/search/SearchFilterInput.tsx

```diff
@@ -62,165 +64,87 @@ const filterNames = [
  * A reusable, autocompleting item search input. This is an uncontrolled input that
  * announces its query has changed only after some delay.
  */
-export default class SearchFilterInput extends React.Component<Props, State> {
-  state: State = { liveQuery: '', filterHelpOpen: false };
-  private textcomplete: Textcomplete;
-  private inputElement = React.createRef<HTMLInputElement>();
-  private debouncedUpdateQuery = _.debounce(this.props.onQueryChanged, 500);
-
-  componentWillUnmount() {
-    if (this.textcomplete) {
-      this.textcomplete.destroy();
-      this.textcomplete = null;
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.searchConfig !== this.props.searchConfig) {
-      this.setupTextcomplete();
-    }
-    if (
-      prevProps.searchQueryVersion !== this.props.searchQueryVersion &&
-      this.props.searchQuery !== undefined
-    ) {
-      this.setState({ liveQuery: this.props.searchQuery });
-    }
-  }
-
-  render() {
-    const { alwaysShowClearButton, placeholder, children, autoFocus } = this.props;
-    const { liveQuery, filterHelpOpen } = this.state;
+export default React.forwardRef(function SearchFilterInput(
```

Comment:
Need to define the types for forwardRef, i.e. `React.forwardRef<HTMLInputElement, Props>`